src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -42,44 +42,6 @@ public static string FormatDateTime(DateTime value)
             return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
         }
 
-        public static bool TryParseAsISO(ReadOnlySpan<char> source, out DateTime value)
```

Comment:
nit: I would prefer if we keep these APIs in the `JsonHelpers` rather than moving the bulk of the implementation to `JsonString`. Any particular reason they were moved?

With the move, most of the implementation ends up calling helper methods anyway that live on `JsonHelpers`.

Reply:
It's about the method name. In the new implementation, the method should not only parse a `JsonString`'s value as ISO but also trim the beginning and ending quotes and unescape characters. I thought the `TryParseAsISO` name doesn't correspond to the actual behavior, so I moved the implementation to `JsonString`.

But if there's no need to trim embedded quotes, we may check if there are backslash characters in the `TryParseAsISO` method and keep `JsonString` as it was before this commit.